python-checks/src/test/resources/checks/tests/testShouldBeSkippedExplicitly.py

```diff
@@ -0,0 +1,162 @@
+        return  # OK, no assertions: might not be an actual test
+    else:
+        print("Hello")
+
+
+def not_a_test():
+    if not external_resource_available():
+        return  # Not named "testXXX"
+    else:
+        assert foo() == 42
+
+
+def helper_test():
+    assert 1 != 2
+
+
+# No issue should be raised if another method is called before the return (except in the "if" condition)
+# As it might be a helper method performing assertions.
+def test_uses_helper():
+    helper_test()
+    if not external_resource_available():
+        return  # OK, call to "helper_test" before
+    assert foo() == 42
+
+
+def test_uses_variable():
+    x = 42
+    if not external_resource_available():
+        return  # Noncompliant {{Skip this test explicitly.}}
+    elif x == 43:
+        x = 43
+    else:
+        x = 44
+    assert foo() == x
+
+def test_uses_variable_binary_left():
+    x = 42
+    if external_resource_available() and x == 42:
+        return  # Noncompliant {{Skip this test explicitly.}}
+    else:
+        x = 43
+    assert foo() == x
+
+def test_uses_variable_binary_right():
+    x = 42
+    if x == 42 and external_resource_available():
+        return  # Noncompliant {{Skip this test explicitly.}}
+    else:
+        x = 43
+    assert foo() == x
+
+def test_uses_variable_2():
+    x = external_resource_available()
+    if not x:
+        return  # Accepted FN
+    assert foo() == 42
+
+
+def test_conditional_skip_not_first_statement():
+    x = 42
+    if x:
+        ...
+    else:
+        ...
+    if not external_resource_available():
+        return  # Accepted FN
+    assert foo() == x
+
+
+# Unittest
+
+class MyTest(unittest.TestCase):
+
+    def test_something(self):
+        if not external_resource_available():
+            return  # Noncompliant {{Skip this test explicitly.}}
+        else:
+            "test".capitalize()
+            self.assertEqual(foo(), 42)
+
+    def test_with_skip(self):
+        if not external_resource_available():
+            self.skipTest("prerequisite not met")
+        else:
+            self.assertEqual(foo(), 42)
+
+    def not_a_test(self):
+        if not external_resource_available():
+            return  # OK, method is not "testXXX"
+        else:
+            self.assertEqual(foo(), 42)  # Assertion doesn't matter as the method is not discoverable by the test runner
+            print("hello")
+
+    def test_no_assertions(self):
+        if not external_resource_available():
+            return  # OK, no assertions in the test (avoid FPs)
+        else:
+            foo()
+
+def test_image_dataset_from_directory_errors(self):
+    if PIL is None:
+      return  # Noncompliant {{Skip this test explicitly.}}
+#     ^^^^^^
+
+    directory = self._prepare_directory(num_classes=3, count=5)
+
+    with self.assertRaisesRegex(ValueError, '`labels` argument should be'):
+      _ = image_dataset.image_dataset_from_directory(
+          directory, labels=None)
+
+
+return
```

Comment:
Is it necessary here ?

Reply:
I believe this is not required anymore.
This was used to cover a specific usecase in the first version of my code, which was using a very different approach : subscribe over RETURN statement instead of FUNCTION DEF.
Having a return out of a function definition was throwing an error.
In any case, I should at least add a comment here to explain why we have this, but I will check and probably remove it as with the new approach this doesn't cover any test cases.